pennylane/transforms/decompose.py

```diff
@@ -135,34 +135,6 @@ def read(self, var):
             """Extract the value corresponding to a variable."""
             return var.val if isinstance(var, jax.extend.core.Literal) else self._env_map[var]
 
-        def stopping_condition(self, op: Operator) -> bool:
-            """Function to determine whether an operator needs to be decomposed or not.
-
-            Args:
-                op (Operator): Operator to check.
-
-            Returns:
-                bool: Whether ``op`` is valid or needs to be decomposed. ``True`` means
-                    that the operator does not need to be decomposed.
-            """
-
-            # If the new graph-based decomposition is enabled,
-            # we don't rely on the has_decomposition attribute.
-            if enabled_graph():
-                return self._stopping_condition(op)
-
-            if not op.has_decomposition:
-                if not self._stopping_condition(op):
-                    warnings.warn(
-                        f"Operator {op.name} does not define a decomposition and was not "
-                        f"found in the target gate set. To remove this warning, add the operator "
-                        f"name ({op.name}) or type ({type(op)}) to the gate set.",
-                        UserWarning,
-                    )
```

Comment:
I've made it so that this warning is raised from `_operator_decomposition_gen`, making this wrapper around the stopping condition unnecessary.